tv_apps/browser/js/search_result.js

```diff
@@ -0,0 +1,240 @@
+  },
+
+  createList: function result_createList() {
+    var list = document.createElement('ul');
+    list.classList.add('result-box');
+    return list;
+  },
+
+  createSearchTemplate: function result_createSearchTemplate() {
+    var template = document.createElement('li');
+    template.classList.add('result-item');
+    var link = document.createElement('div');
+    link.classList.add('result-base');
+    var title = document.createElement('div');
+    title.classList.add('result-title');
+    var url = document.createElement('div');
+    url.classList.add('result-url');
+
+    link.appendChild(title);
+    link.appendChild(url);
+    template.appendChild(link);
+
+    return template;
+  },
+
+  resultUpdate: function result_resultUpdate(filter) {
+    // If an update is already in progress enqueue the following ones
+    if (this.updateInProgress) {
+      this.pendingUpdateFilter = filter;
+      return;
+    } else {
+      this.updateInProgress = true;
+    }
+
+    this.allResult = [];
+    BrowserDB.getBookmarks((function(bookmarks) {
+      this.resultFiltering(filter, bookmarks);
+      BrowserDB.getHistory((function(history) {
+        this.resultFiltering(filter, history);
+        this.populateResults(filter);
+
+        this.updateInProgress = false;
+        var pendingUpdateFilter = this.pendingUpdateFilter;
+        if (pendingUpdateFilter !== null) {
+          this.pendingUpdateFilter = null;
+          this.resultUpdate(pendingUpdateFilter);
+        }
+      }).bind(this));
+    }).bind(this));
+  },
+
+  resultFiltering: function result_resultFiltering(filter, data) {
+    if(( filter == '' ) || ( data == null )) return;
+
+    for( var i = 0 ; i < data.length ; i ++ ) {
+      var current = data[i];
+      var matched = false;
+      if( filter ) {
+        matched = this.matchesFilter(current.uri, filter) ||
+          this.matchesFilter(current.title, filter);
+      }
+      if( matched || !filter ) {
+        var j = 0;
+        for( ; j < this.allResult.length ; j ++ ) {
+          if( this.allResult[j].uri === current.uri ) {
+            break;
+          }
+        }
+        if( j == this.allResult.length ) {
+          this.allResult.push(current);
+        }
+      }
+    }
+  },
+
+  /**
+   * Check if the URI matches the regular expression filter.
+   * @param {String} uri
+   * @param {String} filter
+   * @returns {Boolean}
+   */
+  matchesFilter: function result_matchesFilter(uri, filter) {
+    if( !uri ) return false;
+    return uri.match(new RegExp(filter, 'i')) !== null;
+  },
+
+  populateResults: function result_populateResults(filter) {
+    var list = this.createList();
+    var engine = SearchUtil.getCurrentEngineName();
+    if( engine ) {
+      var result = this.searchResultTemplate.cloneNode(true);
+      var title = result.childNodes[0].childNodes[0];
+      // title.innerHTML = 'Search by ' + engine;
+      title.innerHTML = _('WB_LT_TIPS_S_SEARCH', {value0:engine});
```

Comment:
this should be:

`navigator.mozL10n.setAttributes(title, 'WB_LT_TIPS_S_SEARCH', {value0: engine});`